kernel/src/schema.rs

```diff
@@ -105,6 +106,30 @@ impl StructField {
         self.metadata.get(key.as_ref())
     }
 
+    /// Get the physical name for this field as it should be read from parquet, based on the
+    /// specified column mapping mode.
+    pub fn physical_name(&self, mapping_mode: &ColumnMappingMode) -> DeltaResult<&str> {
+        let name_mapped_name = self
+            .metadata
+            .get(ColumnMetadataKey::ColumnMappingPhysicalName.as_ref());
+        match (mapping_mode, name_mapped_name) {
+            (ColumnMappingMode::None, _) => Ok(self.name.as_str()),
+            (ColumnMappingMode::Name, None) => {
+                Err(Error::MissingData(
+                    format!("No {} key found in field metadata",
+                            ColumnMetadataKey::ColumnMappingPhysicalName.as_ref())
+                ))
+            }
+            (ColumnMappingMode::Name, Some(val)) => match val {
+                MetadataValue::Number(_) => {
+                    Err(Error::generic("{ColumnMetadataKey::ColumnMappingPhysicalName} must be a string in name mapping mode"))
+                }
+                MetadataValue::String(name) => Ok(name),
+            }
```

Comment:
Seems like we could simplify the code a fair bit here?
```suggestion
        let physicalNameKey = ColumnMetadataKey::ColumnMappingPhysicalName.as_ref();
        let name_mapped_name = self.metadata.get(physicalNameKey);
        match (mapping_mode, name_mapped_name) {
            (ColumnMappingMode::None, _) => Ok(self.name.as_str()),
            (ColumnMappingMode::Name, Some(MetadataValue::String(name))) => Ok(name),
            (ColumnMappingMode::Name, invalid) => {
                Err(Error::generic(format!("Missing or invalid {physcialNameKey}: {invalid:?}")))
            }
```